### PR TITLE
fourth hw

### PR DESCRIPTION
**Pull request by bkostanetskyi**

### CODE REVIEW

File: src/main/java/mate/academy/books/controller/BookController.java [hunk 1]
```diff
@@ -33,7 +34,7 @@ public BookDto findById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/controller/BookController.java:37]: should you validate update endpoint also?

File: src/main/java/mate/academy/books/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,14 +1,23 @@
 package mate.academy.books.dto;
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank
```

for String

File: src/main/java/mate/academy/books/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/exception/CustomGlobalExceptionHandler.java]:
```suggestion
```

File: src/main/java/mate/academy/books/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    ) {
+
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/books/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/books/exception/CustomGlobalExceptionHandler.java (hunk 4 above)
Comment (bkostanetskyi) [reply src/main/java/mate/academy/books/exception/CustomGlobalExceptionHandler.java]: I put that in body

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**